### PR TITLE
Support for menus with sections and optional image

### PR DESCRIPTION
**Pull request by ctindel**
This is a port of the 2-year old PR https://github.com/devcows/hugo-universal-theme/pull/111 so it can be merged cleanly originally written by @ringods 

This PR offers:

* dropdown menu with single list of menu items
* dropdown menu with 4 columns of menu items, grouped in sections
* dropdown menu with 2 column wide image, followed by 2 columns of menu items grouped in sections.

Part of https://github.com/devcows/hugo-universal-theme/issues/101

### CODE REVIEW

**Comment by ctindel** [comment]
@GeorgeWL @ryanfox1985 Hi, just wanted to see if we could get this merged at some point, I know it was mentioned recently on the 2 year old PR that you'd accept it if the merge conflicts were removed.

**Comment by ctindel** [comment]
@ryanfox1985 @GeorgeWL any chance of getting this merged?  Wasn't sure if it was blocked behind something else.  Thanks!

**Comment by ryanfox1985** [comment]
I will test during the week.

File: README.md [hunk 1]
```diff
@@ -159,7 +159,106 @@ You can also define the menu items that will appear in the top bar. Edit the `[[
...
+
+The above example shows a reduced version of the *All Pages* menu item from the example site. As you can see,
+we first create menu items with an identifier starting with `section.`. If you create entries like this, these
+will appear as section headers in your drop down. 
+
+Each of these entries contain values for both the `weight` and `post` attribute. The `post` attribute is hijacked
+to indicate in which column a section will be put in. Within a column, the `weight` value is respected to show the
+sections top to bottom.
+
+Use to the unique section identifier (e.g. `section.ap-blog`) as the `parent` value to add a menu item to a specific
```
Comment (ringods) [inline README.md]: `Use to the unique ...`

I see a typo now in my own 2-year old work. 😜

Comment (GeorgeWL) [reply README.md]: ha, removing just one word isn't the worst thing 😛

**Comment by salim-b** [comment]
This PR has some serious deficiencies which should be solved before merging:

- It breaks _navigation highlighting_, originally added in #191 and refactored to rely on `identifier` keys in #211. I currently don't have the time to look into it, but I see you [completely removed the highlighting code in `layouts/partials/nav.html`](https://github.com/devcows/hugo-universal-theme/pull/257/files#diff-e445baf84fcad5164f54b8ea0fa47269L26) and also changed the `identifier` keys from `some_id` to `menu.some_id`...

- The menu in the example site is very confusing since most of the linked subpages don't actually exist. I really dont' see any benefit in mimicking the [upstream theme](https://demo.bootstrapious.com/universal/2-0-2/) _halfway through_.

**Comment by ctindel** [comment]
@salim-b I may be out of my element here since I was just forward-porting someone else's code and it worked ok on my site.  I'm not sure what you mean by navigation highlighting.  I'm running this code on my site as i roll over the top level menu items as well as the sub menu items they do highlight.  Is there another type of highlighting is broken?

That said I didn't realize there was a version of universal theme that had this already unreleased so if you don't think it's worth taking the change maybe we can just drop it.

**Comment by salim-b** [comment]
> I'm running this code on my site as i roll over the top level menu items as well as the sub menu items they do highlight. Is there another type of highlighting is broken?

Yes, the respective top level item should _stay_ highlighted when the corresponding page is visited :wink: 

> I didn't realize there was a version of universal theme that had this already unreleased

I don't really understand what you mean by "already unreleased", but there is ongoing effort to port this whole hugo theme to the new Universal v2 upstream theme in #246

> so if you don't think it's worth taking the change maybe we can just drop it.

Personally I don't need the extended menus, so I probably won't put much work into improving this PR. But since there's no ETA for #246, it might make sense for others to do that work...

**Comment by ryanfox1985** [comment]
shall I close #101 in favour of this PR?

**Comment by ryanfox1985** [comment]
> I'm running this code on my site as i roll over the top level menu items as well as the sub menu items they do highlight. Is there another type of highlighting is broken?

is not working well the highlighting menu.

**Comment by ryanfox1985** [comment]
I updated #111 into #307

**Comment by ryanfox1985** [comment]
closed in favour of #307

**Pull request closed without merging**